okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -831,6 +831,8 @@ public Builder encodedFragment(String encodedFragment) {
     public HttpUrl build() {
       if (scheme == null) throw new IllegalStateException("scheme == null");
       if (host == null) throw new IllegalStateException("host == null");
+      // If we're set to a default port, update it to match the default of the scheme.
+      if (port == 80 || port == 443) port = defaultPort(scheme);
```

Comment:
I _think_ we want to set the port to -1 when setting the port to what would be its default value. That way you can still get HTTPS on 80 and HTTP on 443 as long as you set the scheme first.